File: Haskell/ProjE02.hs
fib :: Int -> Int
fib 0 = 0
fib 1 = 1
fib 2 = 2
fib x = fib ( x - 1) + fib (x - 2)

findAnswer :: Int
findAnswer = sum ([y | y <- takeWhile (<4000000) [fib x | x <- [1..], fib x `mod` 2 == 0]])

main = print findAnswer 

File: CPP/ProjE02.cpp
#include <iostream>
using namespace std;
int main(){
    int a = 1;
    int b = 2;
    int sum = 0;
    for(int c = a+b; b<4000000; c = a+b, a = b, b = c){
        if(b%2 == 0){
            sum+=b;
        }
    }
    cout << "Answer: " << sum << "\n";
    return 0;
}